### DISCUSSION TITLE
Audio handling library

### DISCUSSION THREAD

**Discussion by espkk** [storm-devs/storm-engine#22] [Ideas]
Candidates:

1. FMOD (with linux support?)
2. OpenAL
3. libsoundio
4. your option

Please, share your options, your analysis of the pros and cons

**Comment by egorpugin** [comment]
Again, consider SDL like in other threads (window/input handling). SDL has apis to work with audio. Not sure in main library or in SDL mixer.
https://www.libsdl.org/projects/SDL_mixer/
>  SDL_mixer is a sample multi-channel audio mixer library.
It supports any number of simultaneously playing channels of 16 bit stereo audio, plus a single channel of music, mixed by the popular FLAC, MikMod MOD, Timidity MIDI, Ogg Vorbis, and SMPEG MP3 libraries.

**Comment by Koncord** [comment]
OpenAL is OpenGL-like library, it does not know about audio formats. You will need an external libraries like libogg, libvorbis, etc. for support formats itself. But, I will recommend to use [OpenAL-soft](https://github.com/kcat/openal-soft) or any other implementation instead of phroprietary FMOD or doubtful SDLmixer
